ldms/src/contrib/sampler/Makefile.am

```diff
@@ -0,0 +1,16 @@
+ACLOCAL_AMFLAGS = -I m4
+SUBDIRS = .
+lib_LTLIBRARIES =
+pkglib_LTLIBRARIES =
+
+if ENABLE_IPMIREADER
```

Comment:
I would like to encourage folks to start using the following pattern for conditional subdirs:

if COND_OPT
  MAYBE_OPT = opt
endif
SUBDIRS += $(MAYBE_OPT)

So this one would be:

if ENABLE_IPMIREADER
  MAYBE_IPMIREADER = ipmireader
endif
SUBDIRS += $(MAYBE_IPMIREADER)

When this pattern is followed, automake is able to set DIST_SUBDIRS on its own automatically. This is the recommended pattern from the automake manual:

https://www.gnu.org/software/automake/manual/html_node/Subdirectories-with-AM_005fCONDITIONAL.html#Subdirectories-with-AM_005fCONDITIONAL

Reply:
updated commit to include this @morrone